File: easystack.py
import sys

def main():
    t = int(sys.stdin.readline())
    myStack = []
    readings(t, myStack)
def readings(testCases, myStack):
    while testCases >= 1:
        numbers = sys.stdin.readline().split()
        if numbers[0] == "2" and myStack:
            myStack.pop()
        elif numbers[0] == "3":
            if not myStack:
                sys.stdout.write("Empty!\n")
            else:
                sys.stdout.write(f"{myStack[-1]}\n")
        elif numbers[0] == "1":
            myStack.append(numbers[1])
        testCases -= 1
main()
# =============================================================================================
# import sys
# from io import StringIO

# def main():
#     t = int(input())
#     myStack = []
#     output_buffer = StringIO()

#     for _ in range(t):
#         numbers = input().split()
#         if numbers[0] == "2" and myStack:
#             myStack.pop()
#         elif numbers[0] == "3":
#             output_buffer.write(f"{myStack[-1]}\n" if myStack else "Empty!\n")
#         elif numbers[0] == "1":
#             myStack.append(numbers[1])

#     sys.stdout.write(output_buffer.getvalue())

# main()
# # ===========================================================================================
# import sys
# from collections import deque

# def main():
#     t = sys.stdin.readline()
#     myStack = deque()

#     for _ in range(int(t)):
#         numbers = input().split()
#         if numbers[0] == "2" and myStack:
#             myStack.pop()
#         elif numbers[0] == "3":
#             sys.stdout.write(f"{myStack[-1]}\n" if myStack else "Empty!\n")
#         elif numbers[0] == "1":
#             myStack.append(numbers[1])

# main()

File: easystack.cpp
#include <iostream>
#include <vector>
#include <string>

void readings(int testCases, std::vector<std::string>& myStack);

int main() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(nullptr);

    int t;
    std::cin >> t;
    std::vector<std::string> myStack;
    readings(t, myStack);
    return 0;
}

void readings(int testCases, std::vector<std::string>& myStack) {
    while (testCases >= 1) {
        std::string input;
        std::cin >> input;
        if (input == "2" && !myStack.empty()) {
            myStack.pop_back();
        } else if (input == "3") {
            if (myStack.empty()) {
                std::cout << "Empty!" << '\n';
            } else {
                std::cout << myStack.back() << '\n';
            }
        } else if (input == "1") {
            std::string number;
            std::cin >> number;
            myStack.push_back(number);
        }
        testCases -= 1;
    }
}
